ast/Trees.cc

```diff
@@ -296,42 +296,43 @@ UnresolvedConstantLit::UnresolvedConstantLit(core::LocOffsets loc, ExpressionPtr
 }
 
 ConstantLit::ConstantLit(core::LocOffsets loc, core::SymbolRef symbol, std::unique_ptr<UnresolvedConstantLit> original)
-    : loc(loc), symbol(symbol), original(std::move(original)) {
+    : loc(loc), resolutionScopesOrSymbol(symbol), original(std::move(original)) {
     categoryCounterInc("trees", "resolvedconstantlit");
     _sanityCheck();
 }
 
-optional<pair<core::SymbolRef, vector<core::NameRef>>> ConstantLit::fullUnresolvedPath(core::Context ctx) const {
-    if (this->symbol != core::Symbols::StubModule()) {
```

Comment:
I think that these changes in `fullUnresolvedPath` give me the most pause about this particular change.

We know from the log lines that there are cases where this ENFORCE is wrong: `this->symbol` is set to `StubModule` AND `resolutionScopes` is `nullptr` or `empty()`:

https://go/hkOYxXmc

Reply:
Hm, that's unfortunate.  But it matters how `resolutionScopes` is set: if it's `nullptr`, that's very bad for this PR.

If it's `empty()`, then this PR shouldn't change the behavior, because we'll always have a `resolutionScopes` with a `StubModule`.

Reply:
Yeah I agree. Do you want to add that information to the line that we log when we crash and wait for a bit to confirm? Another option would be to try to reproduce the source of the crash but I think trevor tried that and had to switch off of that task.